lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
Fixing the test cases wouldn't be easy, there are a lot of them and with the above mentioned change we'd have to add both headers and ip. However, I should be able to "fill in" missing attributes.

Also @dead-horse do you think we should throw here if there's no IP? That would essentially mean the request doesn't get processed.